tests/Unit/Evolution/Systems/Burgers/Test_Observe.cpp

```diff
@@ -0,0 +1,272 @@
+  MockRuntimeSystem runner({analytic_solution}, std::move(dist_objects));
+
+  const auto box =
+      db::create<db::AddSimpleTags<Tags::TimeId, Tags::Mesh<1>,
+                                   Tags::Coordinates<1, Frame::Inertial>,
+                                   Burgers::Tags::U>,
+                 db::AddComputeTags<Tags::Time>>(
+          TimeId(true, 0, Slab(0., observation_time).end()), mesh, coords, u);
+
+  observe.run(box, runner.cache(), array_index,
+              std::add_pointer_t<ElementComponent>{});
+
+  // Process the volume and reduction data
+  runner.invoke_queued_simple_action<MockObserverComponent>(0);
+  runner.invoke_queued_simple_action<MockObserverComponent>(0);
+  CHECK(runner.is_simple_action_queue_empty<MockObserverComponent>(0));
+
+  const auto& volume_results = MockContributeVolumeData::results;
+  CHECK(volume_results.observation_id ==
+        observers::ObservationId(Slab(0., observation_time).end()));
+  CHECK(volume_results.subfile_name == "/element_data");
+  CHECK(volume_results.array_component_id ==
+        observers::ArrayComponentId(
+            std::add_pointer_t<ElementComponent>{},
+            Parallel::ArrayIndex<ElementIndex<1>>(array_index)));
+  CHECK(volume_results.in_received_tensor_data.size() == 3);
+  // gcc 6.4.0 gets confused if we try to capture tensor_data by
+  // reference and fails to compile because it wants it to be
+  // non-const, so we capture a pointer instead.
+  const auto check_component =
+      [&element_name, tensor_data = &volume_results.in_received_tensor_data](
+          const std::string& component, const DataVector& expected) noexcept {
+    CAPTURE(*tensor_data);
+    CAPTURE(component);
+    const auto it =
+        alg::find_if(*tensor_data, [name = element_name + "/" + component](
+                                       const TensorComponent& tc) noexcept {
+          return tc.name == name;
+        });
+    CHECK(it != tensor_data->end());
+    if (it != tensor_data->end()) {
+      CHECK(it->data == expected);
+    }
+  };
+  check_component("InertialCoordinates_x", get<0>(coords));
+  check_component("U", get(u));
+  check_component("ErrorU", get(u_error));
+  CHECK(volume_results.received_extents == mesh.extents());
+
+  const auto& reduction_results = MockContributeReductionData::results;
+  CHECK(reduction_results.observation_id ==
+        observers::ObservationId(Slab(0., observation_time).end()));
+  CHECK(reduction_results.subfile_name == "/element_data");
+  CHECK(reduction_results.reduction_names ==
+        std::vector<std::string>{"Time", "NumberOfPoints", "ErrorU"});
+  CHECK(reduction_results.time == observation_time);
+  CHECK(reduction_results.number_of_grid_points ==
+        mesh.number_of_grid_points());
+  // The rest of the RMS calculation is done later by the writer.
+  CHECK(reduction_results.u_error ==
+        alg::accumulate(square(get(u_error)), 0.0));
```

Comment:
I'd prefer to compute this by hand and test for the literal value, with `approx`

Reply:
Unfortunately, while this would work here, the change will not work once this is generalized to other systems in the next PR.  In that PR the variables are filled by writing to the data array of a Variables object.  Variables makes no guarantee about how that array maps to the components of the tensors, so I don't know exactly what I have initialized the tensors to.

Since I would just have to change it back to the version here in the next PR, I would prefer not to change it for this temporary specialized version.

Reply:
we agree with @wthrowe